### PR TITLE
Fix: HTML positioning logic

### PR DESCRIPTION
**Pull request by helenphan0**
Slideout before:
screen shot 2017-09-22 at 6 27 27 pm

Slideout after:
screen shot 2017-09-22 at 6 28 30 pm

Tooltip before (you can scroll to see the rest):
screen shot 2017-09-22 at 6 29 35 pm

Tooltip after:
screen shot 2017-09-22 at 6 30 04 pm

### CODE REVIEW

**Comment by GuruM** [comment]
what does this look like?

**Comment by helenphan0** [comment]
One thing to note: positioning logic isn't perfect when using Full Add Component (ie, there are no other components in the editor and you are adding HTML for the first time).

File: src/editor-actions/Code.js [hunk 1]
```diff
@@ -79,7 +99,7 @@ export default class Code extends React.Component {
     return (
       <div>
         <Menu style={{ ...dropdownStyles, backgroundColor: '#272822' }} className="html-menu">
-          <div style={titleStyles}>{title}</div>
+          <div ref={(el) => this.html = el} style={titleStyles}>{title}</div>
```
Comment (GuruM) [inline src/editor-actions/Code.js]: let's prefix refs with `_` so we always know they're refs

Comment (GuruM) [reply src/editor-actions/Code.js]: perhaps `_menuTitle` for accuracy

File: src/editor-actions/Code.js [hunk 2]
```diff
@@ -10,26 +10,40 @@ import 'brace/theme/monokai';
...
+      leftOffset: 0,
       content: '',
       isSaved: true
     };
   }
 
   componentDidMount() {
     const content = this.props.persistedState.get('content') || '';
+    
+    if (this.html) {
```
Comment (GuruM) [inline src/editor-actions/Code.js]: when wouldn't this be true? if there's a possibility that it isn't going to be true all the time, we should address that case in `componentDidUpdate` (and `shouldComponentUpdate`).

I.e. we can have a state property like `hasCheckedPosition` that is flipped by `setAceEditorPosition()`. In `shouldComponentUpdate` if `!hasCheckedPosition` and `this.html` exists, update. In `componentDidUpdate` if the same condition is true, do `setAceEditorPosition()`.

File: src/editor-actions/Code.js [hunk 3]
```diff
@@ -41,31 +55,37 @@ export default class Code extends React.Component {
...
+    if (!hasRoomToRenderRight) {
+      dropdownStyles.left = leftOffset - 40;
+    }
+
     const titleStyles = secondaryMenuTitleStyle;
 
     const aceEditorProps = Object.assign({}, {
       name: 'code-editor',
-      editorProps: { $blockScrolling: true },
+      editorProps: { $blockScrolling: false },
```
Comment (GuruM) [inline src/editor-actions/Code.js]: hmm if we are currently blocking scrolling, how come we can scroll? 🤔

**Comment by GuruM** [comment]
How come it isn't predictable when using Full Add Component?

**Comment by helenphan0** [comment]
The starting position is different between Full Add Component versus regular add. The Full Add Component is a box div and the regular add is the yellow circular Plus button. The positioning isn't too bad under Full Add Component, it's very slightly offscreen for tooltips/hotspots so I left it as is.

**Comment by GuruM** [comment]
☯️

**Pull request merged**